pylxd/models/_model.py

```diff
@@ -152,7 +152,8 @@ def sync(self, rollback=False):
         for key, val in response.json()['metadata'].items():
             if key not in self.__dirty__ or rollback:
                 setattr(self, key, val)
-                self.__dirty__.remove(key)
+                if key in self.__dirty__:
```

Comment:
I think you should move the fix to the if block above. If I got this right, self.__dirty__ will contain all the attributes that need to be sync so if the key is in self.__dirty__ you set the attr and remove the key 
            if key in self.__dirty__ or rollback:
                setattr(self, key, val)
                self.__dirty__.remove(key)